doc/uml/Class_Diagram

```diff
@@ -0,0 +1,49 @@
+@startuml
+class Data{
+-Idenitifer
+-Content
+}
+
+class Tag{
+-Identifier
+-Sortable Ordinal
++rename()
+}
+
+class Database{
+-Tags[]
+-Data[]
+-Relationships[]
++createData()
++deleteData()
++createTag()
++deleteTag()
+}
+
+class Relationship{
+-Relationship(Data, Tag)
++returnTag()
++returnData()
++changeRelationship()
+}
+
+class Constraint{
+~comparaters()
+}
+
+class Solver{
++filter()
++sort()
+}
+
+Relationship --* Tag
+Relationship --* Data
+
+Database --o Relationship
+Database --o Data
+Database --o Solver
+Database --o Tag
```

Comment:
I think these go the other way around; `Database o-- Relationship`.

See 
![image](https://github.com/ElijahTheRandom/JavaProject321/assets/7352801/cbdc478d-ecf9-45b2-9063-365f2f722394)
The aggregation direction should be toward the `Database`. Should this be composition as well? I don't think we can have any of this stuff without the database: it's sort of a singleton that always exists.